src/mame/seta/simple_st0016.cpp

```diff
@@ -180,55 +187,60 @@ void st0016_state::st0016_io(address_map &map)
  *
  *************************************/
 
-uint32_t st0016_state::latch32_r(offs_t offset)
+uint32_t mayjinsn_state::latch32_r(offs_t offset)
 {
-	if(!offset)
-		m_latches[2] &= ~2;
+	if (!machine().side_effects_disabled())
+	{
+		if (!offset)
+			m_latches[2] &= ~2;
+	}
 	return m_latches[offset];
 }
 
-void st0016_state::latch32_w(offs_t offset, uint32_t data, uint32_t mem_mask)
+void mayjinsn_state::latch32_w(offs_t offset, uint32_t data, uint32_t mem_mask)
 {
-	if(!offset)
+	if (!offset)
 		m_latches[2] |= 1;
 	COMBINE_DATA(&m_latches[offset]);
 	machine().scheduler().synchronize();
```

Comment:
Not an introduced issue, but this isn’t the right way to use `synchronize()` (it isn't actually waiting for synchronisation to do the update.  Same in the other place.